### DISCUSSION TITLE
Any way to give the winbar a darker color?

### DISCUSSION THREAD

**Discussion by ronandalton** [Bekaboo/dropbar.nvim#171] [Q&A] [answered]
Hi, I was wondering if there's some way to give the winbar a darker shade to differentiate it from the other text in the window.

I understand that there is a workaround in the code for an issue (#118) that disables the winbar background color. However, other plugins (eg. fgheng/winbar.nvim) seem to be able to use the proper background color and have colored icons/text, so I'm not entirely sure what the issue is. Is it to do with incorrect colors when clicking in the navbar and as such highlighting a component?

Thank you.

**Comment by Bekaboo** [comment]
Good question, there is an issue in the neovim repo that better describes the problem: https://github.com/neovim/neovim/issues/26037. I am also interested in if there is any way to have a proper winbar background color that is not unnecessarily complex.

> However, other plugins (eg. fgheng/winbar.nvim) seem to be able to use the proper background color and have colored icons/text, so I'm not entirely sure what the issue is

I tried to setup winbar.nvim but it does not even start and set the winbar for me. Have you successfully set it up? If so, could you share you config and a screenshot of how it renders a winbar with background and colorful components?

**Reply by ronandalton** [reply]
Thanks for your response.

Here is the config I use for winbar.nvim:
```lua
return {
  'fgheng/winbar.nvim',
  dependencies = {
    {
      'ColinKennedy/nvim-gps', -- fork of SmiteshP/nvim-gps (which is archived) with some bug fixes
      dependencies = {
        'nvim-treesitter/nvim-treesitter',
      },
      opts = {},
    },
  },
  opts = {
    enabled = true,
  },
}
```

Screenshot:
![image](https://github.com/user-attachments/assets/e5e1ea5c-9127-40a5-8e21-f6ccf1d2487d)

Actually, on taking a closer look at the colors it looks like the winbar color is only applied to the non-text area with that plugin, so not quite what I wanted.

**Reply by Bekaboo** [reply]
Since nvim is merging https://github.com/neovim/neovim/pull/29976, there is no need to combine the highlight groups manually in this plugin -- nvim should handle them properly after merging the patch.

**Reply by ronandalton** [reply]
Nice. I assume at some point in the future this will be possible then once that is merged.

Thanks.

**Comment by Bekaboo** [comment, marked as answer]
You can already have proper background color if you are using nvim 0.11+, see https://github.com/Bekaboo/dropbar.nvim/commit/006566571f7af2edf45c99fb997d550878c661ad

**Reply by Muizzyranking** [reply]
How can i disable it? I do not like it, I want my entire window transparent.
I have tried `vim.api.nvim_set_hl(0, "DropBarCurrentContext", { bg = "#ffffff" })` -- the color is just for testing.

**Reply by Bekaboo** [reply]
@Muizzyranking 
```lua
vim.api.nvim_set_hl(0, "WinBar", {})
vim.api.nvim_set_hl(0, "WinBarNC", {})
```

**Reply by Muizzyranking** [reply]
Thank you, it worked.

**Comment by ronandalton** [comment]
Awesome. I can confirm that the the winbar has the proper background color with nightly now.

![image](https://github.com/user-attachments/assets/c6d9b68f-8733-4877-a8e7-d9ac9a5bc5eb)

Thanks again.
